include/zephyr/bluetooth/audio/bap.h

```diff
@@ -512,7 +512,7 @@ struct bt_bap_unicast_server_cb {
 	 *
 	 * @return 0 in case of success or negative value in case of error.
 	 */
-	int (*config)(struct bt_conn *conn, const struct bt_bap_ep *ep, enum bt_audio_dir dir,
+	int (*config)(const struct bt_conn *conn, const struct bt_bap_ep *ep, enum bt_audio_dir dir,
```

Comment:
IMO making the `conn` pointer `const` is not a good idea. 

It stops the upper layers from using it for anything else (could be that someone wanted to use the `config` callback to trigger e.g. a MCS or TBS discovery).